heartbeat/monitors/wrappers/summarizer/plugerr.go

```diff
@@ -0,0 +1,141 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package summarizer
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/elastic/beats/v7/heartbeat/ecserr"
+	"github.com/elastic/beats/v7/heartbeat/eventext"
+	"github.com/elastic/beats/v7/heartbeat/look"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+// BrowserErrPlugins handles the logic for writing the `error` field
+// for browser monitors, preferentially using the journey/end event's
+// error field for errors.
+type BrowserErrPlugin struct {
+	summaryErrVal  interface{}
+	summaryErr     error
+	stepCount      int
+	journeyEndRcvd bool
+	attempt        int
+}
+
+func NewBrowserErrPlugin() *BrowserErrPlugin {
+	return &BrowserErrPlugin{
+		attempt: 1,
```

Comment:
Do we need this yet another attempt given we already have one in jobSummary which we are using in States plugin?

Reply:
no, but if don't duplicate this here we need to figure out how to communicate across plugins which brings a different sort of complexity. I suppose we could make the summarizer itself more aware of retries as an alternative

Reply:
I did take a stab at it, but I'm not sure it's buggy now, and I'm not sure the effort is worth it at this point. I do think it'd be a nice follow-up improvement.

Reply:
This sparked because of the delay in the retry which considered every event from the agent as an attempt and incrememented in return. This is fixed now with your change and I agree we can do it as followup.